Shift3/boilerplate-client-react
# Routes are now defined declaratively

## Changes
1. All of the routing has been transformed into their declarative counterparts and have been moved into `Routes.tsx`
2. Added a check for the user's value on line 105 in `Routes.tsx`. If the value of user is undefined, it will cause an error when the user logs out.
3. The `NotificationListView` now accepts children. Now, the `UnreadNotifications` and `ReadNotifications` components can be passed into it and rendered within its second column. This was done so that its routing code could be moved to `Routes.tsx`. 

## Purpose
Routes are now declared declaratively in a single file. This makes it easier for developers to understand the boilerplate's feature set.

## Approach
Switched from the jsx based method of declaring routes to the `useRoutes` declarative javascript object way

## Learning
I used the following resources:
- [React Router - Route](https://v5.reactrouter.com/web/api/Route)
- [React Router - useRoutes](https://reactrouter.com/en/main/hooks/use-routes)
- [React Router's useRoutes hook](https://dev.to/dauntless/react-routers-useroutes-hook-38fc)

## Testing Steps

1. Pull in the changes to your local copy of this branch and run it alongside the dj_starter_demo repo.
2. Try using the app as normal. The navigation experience should be the same.
